wallet/src/de/schildbach/wallet/service/CoinJoinService.kt

```diff
@@ -235,26 +262,28 @@ class CoinJoinMixingService @Inject constructor(
 
     private suspend fun updateState(
         mode: CoinJoinMode,
+        timeSkew: Long,
         hasAnonymizableBalance: Boolean,
         networkStatus: NetworkStatus,
         isSynced: Boolean,
         blockChain: AbstractBlockChain?
     ) {
         updateMutex.lock()
         log.info(
-            "coinjoin-old-state: ${this.mode}, ${this.hasAnonymizableBalance}, ${this.networkStatus}, synced: ${this.isSynced} ${blockChain != null}"
+            "coinjoin-old-state: ${this.mode}, ${this.timeSkew / 1000}s, ${this.hasAnonymizableBalance}, ${this.networkStatus}, synced: ${this.isSynced} ${blockChain != null}"
         )
         try {
             setBlockchain(blockChain)
             log.info(
-                "coinjoin-new-state: $mode, $hasAnonymizableBalance, $networkStatus, synced: $isSynced, ${blockChain != null}"
+                "coinjoin-new-state: $mode, ${timeSkew / 1000}s, $hasAnonymizableBalance, $networkStatus, synced: $isSynced, ${blockChain != null}"
             )
             this.networkStatus = networkStatus
             this.hasAnonymizableBalance = hasAnonymizableBalance
             this.isSynced = isSynced
             this.mode = mode
+            this.timeSkew = timeSkew
 
-            if (mode == CoinJoinMode.NONE) {
+            if (mode == CoinJoinMode.NONE || timeSkew > MAX_ALLOWED_TIMESKEW) {
```

Comment:
we will use `timeSkew` to determine if CoinJoin will actually run or not.

`timeSkew` > 2 seconds will give the same result as CoinJoin OFF.